lib/modules/manager/npm/extract/pnpm.ts

```diff
@@ -222,3 +232,108 @@ function getLockedDependencyVersions(
+ * named one (under catalogs:)
+ */
+type PnpmCatalog = { name: string; dependencies: NpmPackageDependency };
+
+export function extractPnpmWorkspaceFile(
+  content: string,
+  packageFile: string,
+): PackageFile | null {
+  logger.trace(`pnpm.extractPnpmWorkspaceFile(${packageFile})`);
+
+  let pnpmCatalogs: Array<PnpmCatalog>;
+  try {
+    pnpmCatalogs = parsePnpmCatalogs(content);
+  } catch {
+    logger.debug({ packageFile }, `Invalid pnpm workspace YAML.`);
+    return null;
+  }
+
+  const extracted = extractPnpmCatalogDeps(pnpmCatalogs);
+
+  if (!extracted) {
+    logger.debug({ packageFile }, 'No dependencies found');
+    return null;
+  }
+
+  logger.debug(extracted, 'Extracted catalog dependencies.');
+
+  return {
+    ...extracted,
+    packageFile,
+  };
+}
+
+function extractPnpmCatalogDeps(
+  catalogs: Array<PnpmCatalog>,
+): PackageFileContent<NpmManagerData> | null {
+  const CATALOG_DEPENDENCY = 'pnpm.catalog';
+
+  const deps: PackageDependency[] = [];
+
+  for (const catalog of catalogs) {
+    for (const [key, val] of Object.entries(catalog.dependencies)) {
+      const depName = parseDepName(CATALOG_DEPENDENCY, key);
+      let dep: PackageDependency = {
+        depType: CATALOG_DEPENDENCY,
+        // TODO(fpapado): for PR discussion, consider how users might be able to
+        // match on specific catalogs for their config.
+        //
+        // For example, we could change depType to `pnpm.catalog.${string}`, so
+        // that users can match use `{matchDepTypes: ["pnpm.catalog.default"]}`,
+        // `{matchDepTypes: ["pnpm.catalog.react17"]}` and so on.
+        //
+        // Another option would be to mess with depName/packageName.
+        //
+        // Is there precedence for something similar?
+        depName,
+        managerData: {
+          // We assign the name of the catalog, in order to know which fields to
+          // update later on.
+          catalogName: catalog.name,
+        },
+      };
+      if (depName !== key) {
+        dep.managerData!.key = key;
+      }
+
+      // TODO: fix type #22198
+      dep = {
+        ...dep,
+        ...extractDependency(CATALOG_DEPENDENCY, depName, val!),
+        prettyDepType: CATALOG_DEPENDENCY,
+      };
+      dep.prettyDepType = CATALOG_DEPENDENCY;
+      deps.push(dep);
+    }
+  }
+
+  return {
+    deps,
+  };
+}
+
+export const pnpmCatalogsSchema = z.object({
+  catalog: z.optional(z.record(z.string())),
+  catalogs: z.optional(z.record(z.record(z.string()))),
+});
+
+function parsePnpmCatalogs(content: string): Array<PnpmCatalog> {
+  const { catalog: defaultCatalogDeps, catalogs: namedCatalogs } =
+    parseSingleYaml(content, { customSchema: pnpmCatalogsSchema });
+
+  return [
+    {
+      name: 'default',
+      dependencies: defaultCatalogDeps ?? {},
+    },
+    ...Object.entries(namedCatalogs ?? {}).map(([name, dependencies]) => ({
+      name,
+      dependencies,
+    })),
+  ];
```

Comment:
```suggestion
  const result = [
    {
      name: 'default',
      dependencies: defaultCatalogDeps ?? {},
    }
  ]

  if (!namedCatalogs) {
    return result;
  }

  for (const [name, dependencies] of Object.entries(namedCatalogs)) {
    result.push({
      name,
      dependencies,
    });
  }

  return result
```
Better readability.
https://github.com/renovatebot/renovate/blob/main/docs/development/best-practices.md#write-understandable-code